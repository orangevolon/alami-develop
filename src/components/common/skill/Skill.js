import { withDOM } from "@hocs";
import "./Skill.scss";

const Skill = ({ badges = [], document, title = "", description = "" }) => {
  const element = document.createElement("section");
  element.className = "skill__section";

  const contentElement = document.createElement("article");
  contentElement.className = "content__article";
  element.appendChild(contentElement);

  const titleElement = document.createElement("h3");
  titleElement.className = "title__h3";
  titleElement.innerText = title;
  contentElement.appendChild(titleElement);

  const descriptionElement = document.createElement("p");
  descriptionElement.className = "description__p";
  descriptionElement.innerText = description;
  contentElement.appendChild(descriptionElement);

  const badgeContainerElement = document.createElement("div");
  badgeContainerElement.className = "badges__div";
  badges.forEach(badge => badgeContainerElement.appendChild(badge));
  element.appendChild(badgeContainerElement);

  return element;
};

export default withDOM(Skill);
