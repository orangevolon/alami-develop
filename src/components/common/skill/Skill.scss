@import "@assets/styles/grid.scss";

.skill__article {
  @include content-grid;

  margin-bottom: 4rem;

  .badges__div {
    position: relative;
    grid-area: badges;
    width: 100%;
    max-width: 400px;
    justify-self: center;
  }

  .title__h3 {
    grid-area: title;
  }

  .description__p {
    grid-area: description;
  }
}
