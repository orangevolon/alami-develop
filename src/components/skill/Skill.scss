.skill__article {
  display: grid;
  grid-template-columns: 1fr 2fr;
  grid-template-rows: auto 1fr;
  grid-template-areas:
    "badges title"
    "badges description";
  column-gap: 4rem;
  row-gap: 1rem;

  margin-bottom: 4rem;

  .badges__div {
    position: relative;
    grid-area: badges;
  }

  .title__h3 {
    grid-area: title;
  }

  .description__p {
    grid-area: description;
  }
}
