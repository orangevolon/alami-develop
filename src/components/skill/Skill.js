import { withDOM } from "../../hocs";
import "./Skill.scss";

const Skill = ({ badges = [], document, title = "", description = "" }) => {
  const element = document.createElement("article");
  element.className = "skill__article";

  const titleElement = document.createElement("h3");
  titleElement.className = "title__h3";
  titleElement.innerText = title;
  element.appendChild(titleElement);

  const descriptionElement = document.createElement("p");
  descriptionElement.className = "description__p";
  descriptionElement.innerText = description;
  element.appendChild(descriptionElement);

  const badgeContainerElement = document.createElement("div");
  badgeContainerElement.className = "badges__div";
  badges.forEach(badge => badgeContainerElement.appendChild(badge));
  element.appendChild(badgeContainerElement);

  return element;
};

export default withDOM(Skill);
