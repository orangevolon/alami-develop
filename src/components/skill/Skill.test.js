import Skill from "./Skill";

describe("Testing components/skill", () => {
  it("Should render the skill correctly", () => {
    const badges = [...new Array(3)].map(() => document.createElement("div"));
    expect(Skill({ badges })).toMatchSnapshot();
  });
});
