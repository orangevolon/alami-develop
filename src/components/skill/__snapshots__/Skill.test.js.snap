// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components/skill Should render the skill correctly 1`] = `
<article
  class="skill__article"
>
  <h3
    class="title__h3"
  />
  <p
    class="description__p"
  />
  <div
    class="badges__div"
  >
    <div />
    <div />
    <div />
  </div>
</article>
`;
